sympy/core/power.py

```diff
@@ -181,6 +181,35 @@ def integer_log(y, x):
     return e, r == 0 and y == 1
 
 
+def binpow(b, e):
+    """
+    Raise base b to integer exponent e > 1 efficiently, using binary
+    representation of e.
+
+    (The reason for not supporting e == 1 is that in that case b itself would
+    be returned, while in all other cases a new instance is computed. Callers
+    should handle the case e == 1 separately, so they can return a clone of b,
+    rather than b itself.)
+    """
+    e = _sympify(e)
```

Comment:
I think that you should either sympify b as well to return a sympified result or not sympify e and get the speed benefit of integer arithmetic (and check e.is_integer in another way, like `int(e) == e`).

Considering that this is in the core it should probably return a sympified result, but that can of course also happen at the return.

Reply:
I see your point regarding speed vs sympifying. Should be one or the other.

Well, as I think more about it, I am beginning to think `binpow` really doesn't belong in `core`. Certainly, I want to be able to use it to compute powers of bases `b` that are "not sympifyable", such as my custom classes.